proxy/src/main/java/com/wavefront/agent/ChannelStringHandler.java

```diff
@@ -69,6 +88,17 @@ protected void channelRead0(ChannelHandlerContext ctx, String msg) throws Except
       }
     }
     // if msg does not match metadata keywords then treat it as a point
+    // use data rate to determine sampling rate
+    // logging includes the source host and port
+    if(logRawDataRate >= 1.0d || (logRawDataRate > 0.0d && RANDOM.nextDouble() < logRawDataRate)) {
+      if(ctx.channel().remoteAddress() != null) {
+        String host = ((InetSocketAddress) ctx.channel().remoteAddress()).getHostName();
+        int port = ((InetSocketAddress) ctx.channel().remoteAddress()).getPort();
```

Comment:
I believe logging remote address and local port would make more sense - remote port is not very useful once the session is over. Maybe write both [local port : remote addr : remote port]?